Sources/Core/ObjectiveCDictionaryExtension.swift

```diff
@@ -162,68 +192,44 @@ extension ObjCFileRenderer {
             }
             let currentResult = "result\(counter)"
             let currentObj = "obj\(counter)"
-            let collection = collectionClass(schema: schema)
-            return ObjCIR.ifElseStmt("\(propIVarName) != nil") {[
-                    "\(collection.name()) *items\(counter) = \(propIVarName);",
+            let itemClass = itemType.isObjCPrimitiveType ? "id" : objcClassFromSchema(param, itemType)
+            return [
+                    "__auto_type items\(counter) = \(propIVarName);",
                     "\(CollectionClass.array.mutableName()) *\(currentResult) = [\(CollectionClass.array.mutableName()) \(CollectionClass.array.initializer())items\(counter).count];",
-                    ObjCIR.forStmt("id \(currentObj) in items\(counter)") { [
-                        ObjCIR.ifStmt("\(currentObj) != (id)kCFNull") { [
-                            createCollection(destCollection: currentResult, processObject: currentObj, collectionSchema: itemType, collectionCounter: counter+1)
-                        ]}
+                    ObjCIR.forStmt("\(itemClass) \(currentObj) in items\(counter)") { [
+                        createCollection(destCollection: currentResult, processObject: currentObj, collectionSchema: itemType, collectionCounter: counter+1)
                     ]},
                     "[\(dictionary) setObject:\(currentResult) forKey:@\"\(param)\"];"
-                ]} {[
-                    "[\(dictionary) setObject:[NSNull null] forKey:@\"\(param)\"];"
-                ]}
+                ].joined(separator: "\n")
         case .map(valueType: .some(let valueType)):
+
             switch valueType {
-            case .map, .array:
-                return
-                    ObjCIR.ifElseStmt("\(propIVarName) != nil") {[
-                        self.renderAddToDictionaryStatement(param, valueType, dictionary)
-                    ]} {[
-                        "[\(dictionary) setObject:[NSNull null] forKey:@\"\(param)\"];"
-                    ]}
-            case .reference(with: _), .oneOf(types: _), .object:
+            case .map, .array, .reference(with: _), .oneOf(types: _), .object:
                 return [
                     "NSMutableDictionary *items\(counter) = [NSMutableDictionary new];",
-                    ObjCIR.forStmt("id key in \(propIVarName)") { [
-                        ObjCIR.ifStmt("[\(propIVarName) objectForKey:key] != (id)kCFNull") { [
-                            "[items\(counter) setObject:[[\(propIVarName) objectForKey:key] dictionaryObjectRepresentation] forKey:key];"
-                        ]}
+                    ObjCIR.forStmt("NSString *key\(counter) in \(propIVarName)") {[
```

Comment:
Will the key always be an NSString?

Reply:
Currently due to the limitations of JSON keys are always String